### PR TITLE
refactor: restore `cargo clippy` check in CI

### PR DESCRIPTION
**Pull request by hamirmahal**

### CODE REVIEW

**Review by hamirmahal — COMMENTED** [review]
I'm happy to move this into its own, separate job, if you want to prevent this from blocking the [`CI / Tests (ubuntu-latest, stable) (pull_request)`](https://github.com/RustAudio/rodio/actions/runs/8448073552/job/23139474937?pr=564) check.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -43,12 +43,8 @@ jobs:
           # Don't use a 'components:' entry--we don't need them with beta/nightly, plus nightly often doesn't have them
           override: true
 
-      # This block can be uncommented to add clippy to CI, but first the existing clippy warnings need to be fixed!
-      # - name: clippy
-      #   run: |
-      #     rustup component add clippy
```
Comment (dvdsk) [inline .github/workflows/ci.yml:49]: do we no longer need to add clippy anymore?

Comment (hamirmahal) [reply .github/workflows/ci.yml:49]: I'm not sure I understand. This pull request adds the `cargo clippy` check back by un-commenting this code out.

**Comment by dvdsk** [comment]
> I'm happy to move this into its own, separate job, if you want to prevent this from blocking the [`CI / Tests (ubuntu-latest, stable) (pull_request)`](https://github.com/RustAudio/rodio/actions/runs/8448073552/job/23139474937?pr=564#step:5:1) check.

No its okay, and thanks for the PR!

**Comment by hamirmahal** [comment]
You're welcome!

File: src/sink.rs
Comment (dvdsk) [inline src/sink.rs:1]: now all that's left is to undo the check and we can get this merged!

**Pull request merged**